web/src/message_store.ts

```diff
@@ -192,30 +194,39 @@ export function get_pm_full_names(user_ids: number[]): string {
     return names.join(", ");
 }
 
-export function set_message_booleans(message: Message): void {
+export function set_message_booleans(message: RawMessage): MessageWithBooleans {
```

Comment:
I could have a separate commit for changing this function to return the new message instead of mutating it.